File: 500_Keyboard_Row/keyboard_row.py
class Solution(object):
    def findWords(self, words):
        """
        :type words: List[str]
        :rtype: List[str]
        """
        rows = [set('qwertyuiop'), set('asdfghjkl'), set('zxcvbnm')]
        ans = []
        for word in words:
            line_cnt = 0
            for row_index in range(len(rows)):
                if set(word) & rows[row_index]: line_cnt += 1
                if line_cnt > 1: break
            if line_cnt == 1: ans.append(word)
        return ans

File: 500_Keyboard_Row/keyboard_row.java
public class Solution {
    public String[] findWords(String[] words) {
        
        int[] row1 = convertToArr("qwertyuiop");
        int[] row2 = convertToArr("asdfghjkl");
        int[] row3 = convertToArr("zxcvbnm");
        
        List<String> ans = new LinkedList<>();
        for (String word : words) {
            if ( all(row1, word) || all(row2, word) || all(row3, word) ) {
                ans.add(word);
            }
        }
        return ans.toArray(new String[0]);
    }
    
    private int[] convertToArr(String s) {
        int[] arr = new int[26];
        for (char c : s.toCharArray()) {
            arr[c - 'a'] = 1;
        }
        return arr;
    }
    
    private boolean all(int[] row, String word) {
        for (char c: word.toLowerCase().toCharArray() ) {
            if ( row [ c - 'a'] == 0) return false;
        }
        return true;
    }
}
